assets/js/components/ChargingPlansSettings.vue

```diff
@@ -133,19 +142,25 @@ export default {
 			const timeISO = time.toISOString();
 			return await api.get(`loadpoints/${this.id}/plan/preview/energy/${energy}/${timeISO}`);
 		},
-		fetchPlan: async function (preview) {
+		fetchPlan: async function () {
 			try {
-				let planRes = null;
-				if (preview && this.socBasedPlanning) {
-					planRes = await this.fetchPlanPreviewSoc(preview.soc, preview.time);
-					this.isPreview = true;
-				} else if (preview && !this.socBasedPlanning) {
-					planRes = await this.fetchPlanPreviewEnergy(preview.energy, preview.time);
-					this.isPreview = true;
-				} else {
+				let nextPlan = this.getNextPlan();
+				console.log(nextPlan);
+
+				let planRes = undefined;
+
+				if (nextPlan.active) {
```

Comment:
Ok, now I understand why you've added the `active` field. But as said other comment. The information about what is the upcoming plan has to come from the server. Doing the active check here for deciding between preview and active is fine.